src/me/corriekay/pokegoutil/controllers/AccountController.java

```diff
@@ -186,33 +201,146 @@ public static void logOn() throws Exception {
+					try {
+						Desktop.getDesktop().browse(new URI(GoogleUserCredentialProvider.LOGIN_URL));
+					} catch (IOException | URISyntaxException e) {
+						System.err.println("Error Opening browser: " + e.toString());
+						return;
+					}
+				} else {
+					UIManager.put("OptionPane.cancelButtonText", "Copy To Clipboard");
+					if(JOptionPane.showConfirmDialog(null, "Please copy this link and paste it into a browser.\nThen, allow the permissions, and copy the code into your clipboard.", "Google Auth", JOptionPane.OK_CANCEL_OPTION, JOptionPane.PLAIN_MESSAGE) == JOptionPane.CANCEL_OPTION){
+						StringSelection ss = new StringSelection(GoogleUserCredentialProvider.LOGIN_URL);
+						Toolkit.getDefaultToolkit().getSystemClipboard().setContents(ss, ss);
+					}
+					UIManager.put("OptionPane.cancelButtonText", "Cancel");
+				}
+				//The user should have the auth code now. Lets get it.
+				authCode = JOptionPane.showInputDialog(null, "Please provide the authentication code", "Google Auth", JOptionPane.PLAIN_MESSAGE);
+			} else {
+				refresh = true;
+			}
+			try {
+				GoogleUserCredentialProvider provider = new GoogleUserCredentialProvider(S_INSTANCE.http);
+				if(refresh) provider.refreshToken(authCode);
+				else provider.login(authCode); 
+				S_INSTANCE.cp = provider;
+				if (config.getBool("login.SaveAuth", false) || checkSaveAuth2()) {
+					if (!refresh)
+						config.setString("login.GoogleAuthToken", provider.getRefreshToken());
+					config.setBool("login.SaveAuth", true);
+				} else {
+					config.delete("login.GoogleAuthToken");
+					config.delete("login.SaveAuth");
+				}
+			} catch (Exception e) {
+				alertFailedLogin2();
+				continue;
+			}
+		}
+		
+	}
+	
+	private static void initOtherControllers(PokemonGo go) {
+		InventoryController.initialize(go);
+		PokemonBagController.initialize(go);
+	}
+	
+	@Deprecated
+	private static void alertFailedLogin() {
+		JOptionPane.showMessageDialog(null, "Unfortunately, your login has failed. Press OK to try again.", "Login Failed", JOptionPane.PLAIN_MESSAGE);
+	}
+	
+	private static void alertFailedLogin2() {
+		//TODO alertFailedLogin
+		return;
+	}
+	
+	@Deprecated
+	private static boolean checkSaveAuth() {
+		UIManager.put("OptionPane.noButtonText", "No");
+		UIManager.put("OptionPane.yesButtonText", "Yes");
+		UIManager.put("OptionPane.okButtonText", "Ok");
+		UIManager.put("OptionPane.cancelButtonText", "Cancel");
+		return JOptionPane.showConfirmDialog(null, "Do you wish to save the password/auth token?\nCaution: These are saved in plain-text.", "Save Authentication?", JOptionPane.YES_NO_OPTION, JOptionPane.PLAIN_MESSAGE) == JOptionPane.OK_OPTION;
+	}
+	
+	private static boolean checkSaveAuth2() {
+		return true;
+	}
+	
+	public static void logOff2() {
+		
+	}
+	
+	//TODO does nothing yet
+	public static void relogNewUser(){
```

Comment:
That function was removed. We just keep logoff() and login(). Think you added it accidently again?